modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/term/contributor/CommerceOrderCommerceDefinitionTermContributor.java

```diff
@@ -119,6 +232,193 @@ public String getFilledTerm(String term, Object object, Locale locale)
 				commerceOrder.getShippingAddress(), locale);
 		}
 
+		if (term.equals(_ORDER_SHIPPING_OPTION)) {
+			CommerceShippingFixedOption commerceShippingFixedOption =
+				_commerceShippingFixedOptionLocalService.
+					fetchCommerceShippingFixedOption(
+						commerceOrder.getCompanyId(),
+						commerceOrder.getShippingOptionName());
+
+			if (commerceShippingFixedOption != null) {
+				return commerceShippingFixedOption.getName(locale);
+			}
+
+			CommerceShippingMethod commerceShippingMethod =
+				commerceOrder.getCommerceShippingMethod();
+
+			if (commerceShippingMethod != null) {
+				CommerceShippingEngine commerceShippingEngine =
+					_commerceShippingEngineRegistry.getCommerceShippingEngine(
+						commerceShippingMethod.getEngineKey());
+
+				if (commerceShippingEngine != null) {
+					List<CommerceShippingOption> commerceShippingOptions =
+						commerceShippingEngine.getCommerceShippingOptions(
+							null, commerceOrder, locale);
+
+					Stream<CommerceShippingOption>
+						commerceShippingOptionsStream =
+							commerceShippingOptions.stream();
+
+					CommerceShippingOption commerceShippingOption =
+						commerceShippingOptionsStream.filter(
+							cso -> cso.getKey(
```

Comment:
We need to find a name for this one like "value", or return directly the value instead of assigning to commerceShippingOption first.